app/controllers/collections_controller.rb

```diff
@@ -131,12 +127,74 @@ def revoke_access
     redirect_to licences_path
   end
 
+  def create_collection_core(name, metadata, owner)
+    name = sanitize_collection_name(name)
+    metadata = update_jsonld_collection_id(metadata, name)
+    uri = metadata['@id']
+    if Collection.find_by_uri(uri).present?  # ingest skips collections with non-unique uri
+      raise ResponseError.new(400), "A collection with the name '#{name}' already exists"
+    else
+      corpus_dir = create_metadata_and_manifest(name, convert_json_metadata_to_rdf(metadata))
+      # Create the collection without doing a full ingest since it won't contain any item metadata
+      collection = check_and_create_collection(name, corpus_dir)
+      collection.owner = owner
+      collection.save
+      "New collection '#{name}' (#{uri}) created"
+    end
+  end
+
+  def web_create_collection
```

Comment:
This method is too big and doing too many things. It should be decomposed into methods which: sanitise input, check for require fields, validate fields, and lastly creates the collection.